packages/tools/viewer-alpha/test/apps/web/index.html

```diff
@@ -109,8 +115,12 @@
         <button class="toggle-engine-button" onclick="onToggleEngine()">Toggle Engine</button>
 
         <button id="anchor-button" class="toggle-hotspot-button" onclick="onToggleHotSpot()">Toggle Hot Spot</button>
+        <button id="inspector-button" class="toggle-inspector-button" onclick="onToggleInspector()">Toggle Inspector</button>
         <script type="module" src="/packages/tools/viewer-alpha/src/index.ts"></script>
-        <script>
+        <script type="module">
+            import "core/Engines/WebGPU/Extensions/engine.query";

```

Comment:
Is something in the Insepctor path supposed to import this automatically? If I don't have this, the stats part of inspector fails.

Reply:
I don't remember testing the inspector with webgpu, so it might be something we have missed. @Popov72 - should this be included in the main WebGPUEngine module?

Reply:
Would be great if possible ? or the time should not be displayed if not present ?

Reply:
Yes, I think we should add it to the main engine. Can you add the `import "./engine.query";` line in `webgpuEngine.ts` as part as your PR @ryantrem?

Reply:
I took a look but I don't see how to do this without introducing a circular dependency. `engine.query` already imports `webgpuEngine` to both augment its prototype and having those augmentations access state from `webgpuEngine` itself (not its base classes). Why do we even have these things (engine.query, engine.alpha, etc.) in separate files if they are statically imported into webgpuEngine anyway? Can we just move the code from engine.query into webgpuEngine?

Reply:
In fact, the occlusion query code should be part of `ThinWebGPUEngine`. I created a PR for it:

https://github.com/BabylonJS/Babylon.js/pull/15830

Reply:
merged